crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/stateless/StatelessAssessmentService.java

```diff
@@ -0,0 +1,234 @@
+        final var childWeightings = criteriaEntry.getAssessmentCriteriaChildWeightings();
+
+        final var children = childGroupings
+                .entrySet()
+                .stream().map(childGroup -> {
+                    var weighting = childWeightings
+                            .stream()
+                            .filter(weightingEntity -> {
+                                final AgeRange grouping = childGroup.getKey();
+                                return weightingEntity.getLowerAgeRange() == grouping.getLowerLimit() &&
+                                        weightingEntity.getUpperAgeRange() == grouping.getUpperLimit();
+                            })
+                            .findFirst();
+                    return new ApiAssessmentChildWeighting()
+                            .withChildWeightingId(weighting.get().getId())
+                            .withNoOfChildren(childGroup.getValue());
+                }).collect(Collectors.toList());
+        final var childIds = children.stream().map(ApiAssessmentChildWeighting::getChildWeightingId).collect(Collectors.toSet());
+        final var zeroes = childWeightings
+                .stream()
+                .filter(x -> !childIds.contains(x.getId()))
+                .map(x -> new ApiAssessmentChildWeighting()
+                        .withChildWeightingId(x.getId())
+                        .withNoOfChildren(0)).toList();
+        // Service requires non-present groups to be filled in with zero values
+        children.addAll(zeroes);
+        return children;
+    }
+
+    private static BigDecimal incomeTotal(@NotNull List<Income> incomes) {
+        var values = incomes.stream().map(income -> {
+            final var applicant = income.getApplicant();
+
+            var incomevalue = BigDecimal.valueOf(applicant.getFrequency().getWeighting()).multiply(applicant.getAmount());
+            final var partner = income.getPartner();
+            if (partner != null) {
+                incomevalue = incomevalue.add(BigDecimal.valueOf(partner.getFrequency().getWeighting()).multiply(partner.getAmount()));
+            }
+            return setStandardScale(incomevalue);
+        });
+
+        return values.reduce(BigDecimal.ZERO, BigDecimal::add);
+    }
+
+    private static BigDecimal outgoingsTotal(@NotNull List<Outgoing> outgoings) {
+        var values = outgoings.stream().map(outgoing -> {
+            final var applicant = outgoing.getApplicant();
+
+            var incomevalue = BigDecimal.valueOf(applicant.getFrequency().getWeighting()).multiply(applicant.getAmount());
+            final var partner = outgoing.getPartner();
+            if (partner != null) {
+                incomevalue = incomevalue.add(BigDecimal.valueOf(partner.getFrequency().getWeighting()).multiply(partner.getAmount()));
+            }
+            return setStandardScale(incomevalue);
+        });
+
+        return values.reduce(BigDecimal.ZERO, BigDecimal::add);
+    }
```

Comment:
The challenge is that this structure is different. It's just adding things up. It feels simpler to 'duplicate' rather than try to bend this structure into another one just to add up some numbers

Reply:
After talking to David Read, I have now implemented this by calling the original code so that we can call more of it via doAssessment() at a later date.